user_tools/src/spark_rapids_dataproc_tools/validation_scripts/metadata_validation.py

```diff
@@ -0,0 +1,196 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+from pyspark import SparkContext
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col, min, max, avg, stddev, countDistinct, when, asc, round
+
+import time
+import fnmatch
+from pyspark.sql.types import DoubleType
+
+def validation(spark, args):
+
+    result = top_level_metadata(spark, args.format, args.t1, args.t2, args.t1p, args.t2p, args.f)
+    print('-------top level metadata info-------')
+    print(result.show())
+
+    # A result table with the same PK but different values for that column(s)
+    result = metrics_metadata(spark, args.format, args.t1, args.t2, args.t1p, args.t2p, args.pk, args.i, args.e, args.f, args.p)
+    print('-------metadata diff info-------')
+    print(result.show())
+
+    start_time = time.time()
+    print('------------run validation success-----')
+
+def top_level_metadata(spark, format, t1, t2, t1p, t2p, f):
+    if format in ['parquet', 'orc', 'csv']:
+        print('todo')
+    elif format == "hive":
+        results = []
+        table_names = [t1, t2]
+        where_clause = ''
+        if t1p != 'None' and f != 'None':
+            where_clause = f" where {t1p} and {f}"
+        elif t1p != 'None':
+            where_clause = f" where {t1p}"
+        elif f != 'None':
+            where_clause = f" where {f}"
+        for table_name in table_names:
+            sql = f'select * from {table_name}'
+            sql += where_clause
+            df = spark.sql(sql)
+            row_count = df.count()
+            col_count = len(df.columns)
+            results.append((table_name, row_count, col_count))
+        resultsDF = spark.createDataFrame(results, ["TableName", "RowCount", "ColumnCount"])
+        return resultsDF
+
+def generate_metric_df(spark, table_DF, i, t1):
+    result = None
+    agg_functions = [min, max, avg, stddev, countDistinct]
+    # if not specified any included_columns, then get all numeric cols
+    metrics_cols = i
+    if i in ['None', 'all']:
+        metrics_cols = [c.name for c in table_DF.schema.fields if
+                        any(fnmatch.fnmatch(c.dataType.simpleString(), pattern) for pattern in ['*int*', '*decimal*', '*float*', '*double*'])]
```

Comment:
Just wanted to check if column with string schema will be handled separately in this PR or in a follow on PR?